programmers/20210506/문자열-다루기.js

```diff
@@ -0,0 +1,4 @@
+a = ["1234", "a234", "a", "1", "aq", "12", "as2", "123", "12345", "asd23", "asdf12", "123456", "1234567", "asd1234"]
+for (let i = 0; i < a.length; i++) {
+    console.log((!isNaN(a[i]) && (a[i].length === 4 || a[i].length === 6)))
+}
```

Comment:
잘 푸셨네요 isNaN이 안되는 이유는 앞서 순기님이 말씀해주셨듯이 11.1인경우를 문자가 아닌 숫자로 받아들여 그렇습니다. 그렇기 때문에 isNaN이 아닌 정규표현식을 이용해서 풀이하시면 됩니다.
```/^[0-9]+$/``` 
``` ^ ``` 문장의 시작
``` [0-9] ``` 숫자
``` + ``` 한개거나 그보다 많다
``` & ``` 문장의 끝

그 후 test메소드를 통해 true 또는 false를 반환받습니다.   

```suggestion
function solution(s) {
    if((s.length === 4 || s.length === 6) && /^[0-9]+$/.test(s)) {
        return true;
    }
    return false;
}
```